Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionSidebarCocoa.mm

```diff
@@ -215,19 +304,86 @@
 
 void WebExtensionSidebar::setSidebarPath(std::optional<String> sidebarPath)
 {
+    // If we previously did not have a path override in this sidebar, then the browser is currently holding a reference to
+    // our parent's WebView in correspondence to this sidebar. Similarly, if we previously had an override in this sidebar
+    // and it has just been cleared, the browser will hold a reference to this sidebar's WebView. Thus, in both of these cases,
+    // we need to tell the browser to ask us for a new reference of the appropriate specificity. However, these conditions do not
+    // apply to the default sidebar, as it must always have a WebView -- so if we are the default sidebar, never signal the browser
+    // to ask for a new WebView reference.
+    bool needToUpdateWebView = (static_cast<bool>(sidebarPath) != static_cast<bool>(m_sidebarPathOverride)) && !isDefaultSidebar();
+
     if (!sidebarPath && isDefaultSidebar() && extensionContext())
         m_sidebarPathOverride = getDefaultSidebarPathFromExtension(extensionContext().value()->extension());
     else
         m_sidebarPathOverride = sidebarPath;
 
-    propertiesDidChange();
+    if (!m_sidebarPathOverride)
+        m_webView = nil;
+
+    if (needToUpdateWebView)
+        webViewWasUpdated();
+    else
+        reloadWebView();
+}
+
+void WebExtensionSidebar::willOpenSidebar()
+{
+    ASSERT(webView());
+}
+
+void WebExtensionSidebar::willCloseSidebar()
+{
+    // For now, we will only set tab-specific sidebars to nil on close
+    if (m_tab && m_webView)
+        m_webView.clear();
+}
+
+void WebExtensionSidebar::reloadWebView()
+{
+    if (!m_webView)
+        return;
+
+    auto url = URL { extensionContext().value()->baseURL(), sidebarPath() };
+    [m_webView loadRequest:[NSURLRequest requestWithURL:url]];
 }
 
 WKWebView *WebExtensionSidebar::webView()
 {
+    std::optional<Ref<WebExtensionContext>> maybeContext;
+    if (!opensSidebar() || !(maybeContext = extensionContext()))
+        return nil;
+    Ref<WebExtensionContext> context = WTFMove(maybeContext.value());
+
+    // If we have no particular path configured for this sidebar, just give the parent's WebView (whose path will apply to this sidebar)
+    if (std::optional<Ref<WebExtensionSidebar>> parentRef; !m_sidebarPathOverride && !isDefaultSidebar() && (parentRef = parent()))
+        return parentRef.value()->webView();
```

Comment:
AH, I think I see now. But we should just always own the web view per sidebar. Sharing a web view will lead to more problems than it is worth.